.github/workflows/cd.yml

```
@@ -0,0 +1,161 @@
+      - master
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_OWNER: lecafecloud
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+    strategy:
+      matrix:
+        component: [backend, frontend]
+    outputs:
+      image-tag: ${{ steps.meta.outputs.tag }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v5
+        with:
+          fetch-depth: 0
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Generate image metadata
+        id: meta
+        run: |
+          git fetch --tags
+
+          GIT_TAG=$(git describe --tags --abbrev=0 2>/dev/null)
+          TAG=${GIT_TAG#v}
+
+          echo "tag=${TAG}" >> $GITHUB_OUTPUT
+          echo "Image will be tagged as: ${TAG}"
+
+      - name: Log in to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build and push ${{ matrix.component }} Docker image
+        uses: docker/build-push-action@v6
+        with:
+          context: ./${{ matrix.component }}
+          file: ./${{ matrix.component }}/Dockerfile
+          push: true
+          tags: |
+            ${{ env.REGISTRY }}/${{ env.IMAGE_OWNER }}/terrascope-${{ matrix.component }}:${{ steps.meta.outputs.tag }}
+            ${{ env.REGISTRY }}/${{ env.IMAGE_OWNER }}/terrascope-${{ matrix.component }}:latest
+          cache-from: type=gha,scope=${{ matrix.component }}
+          cache-to: type=gha,mode=max,scope=${{ matrix.component }}
+          provenance: false
+
+      - name: Scan ${{ matrix.component }} image for vulnerabilities
+        uses: aquasecurity/trivy-action@0.28.0
+        with:
+          image-ref: ${{ env.REGISTRY }}/${{ env.IMAGE_OWNER }}/terrascope-${{ matrix.component }}:${{ steps.meta.outputs.tag }}
+          format: "sarif"
+          output: "trivy-${{ matrix.component }}-results.sarif"
+          severity: "CRITICAL,HIGH"
+
+      - name: Upload Trivy scan results
+        uses: github/codeql-action/upload-sarif@v4
+        if: always()
+        with:
+          sarif_file: "trivy-${{ matrix.component }}-results.sarif"
+          category: "trivy-${{ matrix.component }}-production"
+
+  update-k8s-manifests:
+    needs: deploy
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v5
+        with:
+          token: ${{ secrets.GITOPS_TOKEN }}
+          fetch-depth: 0
+
+      - name: Update Backend deployment manifest
+        run: |
+          IMAGE_TAG="${{ needs.deploy.outputs.image-tag }}"
+          NEW_IMAGE="${{ env.REGISTRY }}/${{ env.IMAGE_OWNER }}/terrascope-backend:${IMAGE_TAG}"
+
+          sed -i "s|image: ${{ env.REGISTRY }}/${{ env.IMAGE_OWNER }}/terrascope-backend:.*|image: ${NEW_IMAGE}|g" \
```

ton approche est tres bien mais je partage ca parce que ca fait moins hacky. c'est pas une demande de changement tkt
```suggestion
          yq -i '.spec.template.spec.containers[0].image = strenv(NEW_IMAGE)' infra/k8s/deployments/backend.yml \
```

---

J'aime bien ta suggestion, je vais modifier c'est plus compréhensible ton snippet